### ISSUE TITLE
Does not handle case where browser reformats the html

### ISSUE THREAD

**Issue by slorber** [lovasoa/react-contenteditable#18]
Sometimes when you do `node.innerHtml = stateHtml`, the browser do reformat the html, so you may end up with a `stateHtml` in your state that is different from `node.innerHtml`

It was a problem in my app because it could lead the component to always return true to `shouldComponentUpdate`, making the component always re-render (including caret jump problems etc).

In my app I fixed it with something like that:

```javascript
    componentDidUpdate: function() {
        const node = ReactDOM.findDOMNode(this);
        const html = this.props.html;

        // Bypass VDOM diff by React and updates the dom if it is not correct
        // See http://stackoverflow.com/a/27255103/82609
        if ( node.innerHTML !== html ) {
            node.innerHTML = this.props.html;

            // If DOM node is still different after that, it's because the browser decided to reformat it!
            // we emit the reformatting as a change so that we are sync with the store!
            // (otherwise shouldComponentUpdate may not kick in next time, leading to re-render!)
            const finalInnerHtml = node.innerHTML;
            if ( finalInnerHtml !== html ) {
                this.props.onChange(finalInnerHtml);
            }
        }
    },
```

Just reporting that problem for documentation purpose, because I can't make a PR as your onChange callback takes an event and I don't have one in this case.

**Comment by geoffliu** [comment]
+1 on this issue, I'm running into it as well. @slorber and @Treora I published #26, which seems to fix the problem in a more straightforward way. What do you guys think?

**Comment by Treora** [comment]
Should be fixed since #27.
